src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -278,12 +288,11 @@ static String possiblyTrimStdio(Collection<CaseResult> results, boolean keepLong
     private static final int HALF_MAX_SIZE = 500;
     private static final int HALF_MAX_FAILING_SIZE = 50000;
     private static int halfMaxSize(Collection<CaseResult> results) {
-        for (CaseResult result : results) {
-            if (result.errorStackTrace != null) {
-                return HALF_MAX_FAILING_SIZE;
-            }
-        }
-        return HALF_MAX_SIZE;
+        return hasFailures(results) ? HALF_MAX_FAILING_SIZE : HALF_MAX_SIZE;
+    }
+
+    private static boolean hasFailures(Collection<CaseResult> results) {
+        return results.stream().anyMatch(r -> r.errorStackTrace != null);
```

Comment:
IIUC this is just switching to `Stream`, but not changing method behavior.

Reply:
Yeah, that's correct. If you'd prefer, we can stick to the existing for-loop style.

Reply:
No opinion (nor am I a maintainer), just noting while trying to follow what the code changes do.